src/PowerShellEditorServices.Host/CodeLens/ReferencesCodeLensProvider.cs

```diff
@@ -50,7 +49,7 @@ public ReferencesCodeLensProvider(EditorSession editorSession)
         /// </summary>
         /// <param name="scriptFile">The PowerShell script file to get code lenses for.</param>
         /// <returns>An array of CodeLenses describing all functions in the given script file.</returns>
-        public CodeLens[] ProvideCodeLenses(ScriptFile scriptFile)
+        public async Task<CodeLens[]> ProvideCodeLenses(ScriptFile scriptFile)
```

Comment:
You made this function `async`, but didn't add any `await` statements so `async` isn't needed here.

Reply:
I had to change it here as well due to the signature change in the interface. The `PesterCodeLensProvider` properly implements async-await, which is what counts

Reply:
Interesting! Typically the compiler yells if you have an `async` but no `await` but I guess since the interface requires it, the compiler doesn't yell.

Reply:
Yep, it emits a warning though that this function won't run async as expected

Reply:
You could changed the Function to return:

```csharp
return await Task.FromResult(acc.ToArray());
```

to make it happy I think.

Reply:
I don't think this is a good idea because you might as well just suppress the warning. The compiler is happy, it is just kind to let us know that this other implementation will actually run synchronously despite the async declaration, Task.FromResult will still make it run sync but add unnecessary overhead. The warning is good to have because it is a way of tracking possible future improvement and also letting methods being called from this method know that they will actually need to adapt on a higher level if someone made them truly async as well